examples/common/retry_handling.py

```diff
@@ -0,0 +1,22 @@
+"""Example to handle retry for clients for HTTP status/errors."""
+
+from nisystemlink.clients.core import retry
+from nisystemlink.clients.file import FileClient
+
+# Retry on Too Many Requests status or any Exception
+WHEN_CLAUSE = retry.when.status(429) | retry.when.raises(Exception)

```

Comment:
As I said in my other comment, we'd like to have a default pretty good retry config that is useful in most cases that is simple to add and can be updated as we find cases to include. Could you take a crack at making one? 

Retrying 502 and 503 would be good, and 429 as you have. I'm not sure what Exception cases you're trying to retry, did you have something in mind? I would start without that. We definitely want to stop after a number of retries, but I think we can leave the delay off and just rely on the count of retries plus backoff to get to the time we want.